api/src/lang/parse.rs

```diff
@@ -88,123 +103,117 @@ fn parse_value_source(input: &str) -> IResult<&str, ValueSource> {
     ))(input)
 }
 
-fn parse_read(input: &str) -> IResult<&str, Instr> {
+fn parse_read(input: &str) -> IResult<&str, Instr, VerboseError<&str>> {
     // input is remaining stuff to parse
     // tuple is output values, we throw away the first two because that's
     // "Read" and the whitespace delim
     // >>> Read RX0
-    let (input, (_, _, reg)) =
-        tuple((instr("Read"), arg_delim, reg_ident))(input)?;
+    let (input, _) = context("Read", instr("Read"))(input)?;
```

Comment:
Why not put the `context` call inside `instr()` instead, so you don't have to repeat it for every instruction?

Reply:
Big brain